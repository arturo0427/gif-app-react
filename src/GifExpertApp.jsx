import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);


    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;

        // setCategories(categories.concat(category));

        setCategories([newCategory, ...categories]);
    };

    return (

        <>
            {/* Titulo */}
            <h1>Gif Expert App</h1>
            {/* Input */}

            <AddCategory
                // setCategories={setCategories} 
                onNewCategory={event => onAddCategory(event)}

            />

            {/* Listado de gifs */}
            {/* <button onClick={() => onAddCategory('Futurama')}>Agregar</button> */}
            <ol>
                {categories.map((category) =>
                (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                )
                )}
            </ol>
        </>
    )
}
