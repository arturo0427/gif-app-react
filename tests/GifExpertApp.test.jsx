import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Prueba del componente <GifExpertApp />', () => {

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el valor inicial de One Punch', () => {
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');

        expect(input.value).toBe('a');
    });

});