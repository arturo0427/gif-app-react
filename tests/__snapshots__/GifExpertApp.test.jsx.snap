// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h2>
      Cargando...
    </h2>
    <h3>
      One Punch
    </h3>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
